### PR TITLE
test/fix: e2e swap quotes test in one for zero direction (right); some bug fixes

### PR DESCRIPTION
**Pull request by p0mvn**
## What is the purpose of the change

E2E quotes test for swapping one for zero (right). Note that it requires a localosmosis change to increase the amounts of genesis tokens

There are bug fixes. Namely,
- bound tick estimation in edge cases
- silencing panics with inverting large prices (still needs to be fixed to be displayed correctly)
- correctly setting slippage when creating inactive position

Relevant osmosis PR: https://github.com/osmosis-labs/osmosis/pull/5790

## Brief Changelog
- added more simple and edge case tests
- added a complex test of swapping to far away tick that is near max spot price along a range of initialized positons

### CODE REVIEW

File: packages/stores/src/account/osmosis/index.ts [hunk 1]
```diff
@@ -627,24 +626,8 @@ export class OsmosisAccountImpl {
...
-          quoteCoin &&
-          // full tolerance if 0 sqrt price so no positions
-          !queryPool.concentratedLiquidityPoolInfo?.currentSqrtPrice.isZero()
-            ? new Dec(quoteCoin.amount)
-                .mul(new Dec(1).sub(new Dec(maxSlippage).quo(new Dec(100))))
-                .truncate()
-                .toString()
-            : "0";
+        const token_min_amount0 = "0";
+        const token_min_amount1 = "0";
```
Comment (p0mvn) [inline packages/stores/src/account/osmosis/index.ts:630]: Temporary change. Had tests broken because of this. Need to figure out

File: packages/stores/src/queries/concentrated-liquidity/tick-data-provider.ts [hunk 2]
```diff
@@ -23,8 +23,8 @@ export class ConcentratedLiquidityPoolTickDataProvider
 
   constructor(
     protected readonly queryLiquiditiesNetInDirection: ObservableQueryLiquiditiesNetInDirection,
-    protected readonly nextTicksRampMultiplier = new Int(8),
-    protected readonly maxNumRequeriesPerDenom = 9
+    protected readonly nextTicksRampMultiplier = new Int(20),
+    protected readonly maxNumRequeriesPerDenom = 100
```
Comment (p0mvn) [inline packages/stores/src/queries/concentrated-liquidity/tick-data-provider.ts:27]: temporary: need to experiment and tweak this more

**Comment by jonator** [comment]
@p0mvn https://github.com/osmosis-labs/osmosis-frontend/pull/1775

This may fix things for you

File: packages/pools/src/concentrated.ts [hunk 3]
```diff
@@ -251,16 +251,18 @@ export class ConcentratedLiquidityPool implements BasePool, RoutablePool {
...
-      beforeSpotPriceOutOverIn: new Dec(1).quoTruncate(
-        beforeSpotPriceInOverOut
-      ),
+      beforeSpotPriceOutOverIn: invertIfNonZero(beforeSpotPriceInOverOut),
       afterSpotPriceInOverOut,
-      afterSpotPriceOutOverIn: new Dec(1).quoTruncate(afterSpotPriceInOverOut),
+      afterSpotPriceOutOverIn: invertIfNonZero(afterSpotPriceInOverOut),
       effectivePriceInOverOut,
-      effectivePriceOutOverIn: new Dec(1).quoTruncate(effectivePriceInOverOut),
+      effectivePriceOutOverIn: invertIfNonZero(effectivePriceInOverOut),
```
Comment (p0mvn) [inline packages/pools/src/concentrated.ts:265]: link

File: packages/stores/src/__tests_e2e__/stop-osmosisd.sh
Comment (jonator) [inline packages/stores/src/__tests_e2e__/stop-osmosisd.sh:1]: Please revert

File: packages/stores/src/__tests_e2e__/start-osmosisd.sh
Comment (jonator) [inline packages/stores/src/__tests_e2e__/start-osmosisd.sh:1]: Please revert

File: packages/stores/src/__tests_e2e__/global-setup.ts
Comment (jonator) [inline packages/stores/src/__tests_e2e__/global-setup.ts:1]: Please revert

File: packages/stores/src/__tests_e2e__/test-env.ts [hunk 4]
```diff
@@ -30,8 +31,8 @@ export const chainId = "localosmosis";
 
 export const TestChainInfos: (ChainInfo & Chain)[] = [
   {
-    rpc: "http://127.0.0.1:26657",
-    rest: "http://127.0.0.1:1317",
+    rpc: "http://207.154.252.194:26657",
+    rest: "http://207.154.252.194:1317",
```
Comment (jonator) [inline packages/stores/src/__tests_e2e__/test-env.ts:35]: Please revert

File: packages/stores/src/account/__tests_e2e__/swap-exact-in-positions.spec.ts [hunk 5]
```diff
@@ -47,38 +70,408 @@ describe("Test Swap Exact In - Concentrated Liquidity", () => {
...
-      quote.amount,
-      tokenOutCurrency
-    );
+        console.log(
+          "step " + i,
+          tokenInAmountRemainingDec.toString(),
+          amountInConsumed.toString(),
+          amountOutComputed.toString(),
+          feeChargeTotal.toString()
+        );
```
Comment (jonator) [inline packages/stores/src/account/__tests_e2e__/swap-exact-in-positions.spec.ts:137]: Please remove all logs

File: packages/stores/src/queries/concentrated-liquidity/tick-data-provider.ts [hunk 6]
```diff
@@ -45,6 +45,10 @@ export class ConcentratedLiquidityPoolTickDataProvider
       currentTickLiquidity: pool.currentTickLiquidity,
     });
 
+    // console.log("boundTickIndex", boundTickIndex.toString())
+    // console.log("nextTicksRampMultiplier", this.nextTicksRampMultiplier.toString())
+    // console.log("maxNumRequeriesPerDenom", this.maxNumRequeriesPerDenom.toString())
+
```
Comment (jonator) [inline packages/stores/src/queries/concentrated-liquidity/tick-data-provider.ts:51]: Please clean this up

**Review by jonator — CHANGES_REQUESTED** [review]

File: packages/stores/src/queries/concentrated-liquidity/tick-data-provider.ts [hunk 7]
```diff
@@ -23,7 +23,7 @@ export class ConcentratedLiquidityPoolTickDataProvider
 
   constructor(
     protected readonly queryLiquiditiesNetInDirection: ObservableQueryLiquiditiesNetInDirection,
-    protected readonly nextTicksRampMultiplier = new Int(8),
+    protected readonly nextTicksRampMultiplier = new Int(9),
```
Comment (p0mvn) [inline packages/stores/src/queries/concentrated-liquidity/tick-data-provider.ts:26]: Confirmed that this is the smallest possible to allow all tests to pass (and cover entire tick range)

**Review by jonator — CHANGES_REQUESTED** [review]
Please fix build errors

File: packages/math/src/pool/concentrated/tick.ts [hunk 8]
```diff
@@ -206,9 +208,23 @@ export function estimateInitialTickBound({
...
+        estimate = tickToSqrtPrice(currentTick.sub(constantTickEstimateMove));
+      } else {
+        // price is increasing so move estimate up
+        estimate = tickToSqrtPrice(currentTick.add(constantTickEstimateMove));
+      }
+    } else {
+      estimate = currentSqrtPrice.add(
+        new Dec(tokenIn.amount).quo(currentTickLiquidity)
+      );
+    }
```
Comment (p0mvn) [inline packages/math/src/pool/concentrated/tick.ts:227]: here
